src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3869,32 +3869,36 @@ void MarkTypeForDynamicallyAccessedMembers (ref ReflectionPatternContext reflect
 			void MarkMethodsFromReflectionCall (ref ReflectionPatternContext reflectionContext, TypeDefinition declaringType, string name, BindingFlags? bindingFlags, int? parametersCount = null)
 			{
 				bool foundMatch = false;
-				foreach (var method in declaringType.Methods) {
-					var mname = method.Name;
+				do {
+					foreach (var method in declaringType.Methods) {
```

Comment:
Note that reflection will not return private members of base type - ever. So this should also ignore private members on base class. Take a look at `MarkMethodsOnTypeHierarchy` it already implements all that logic. Maybe we could merge the two now?